guides/release/tutorial/12-service-injection.md

```diff
@@ -0,0 +1,531 @@
+<!-- Heads up! This is a generated file, do not edit directly. You can find the source at https://github.com/ember-learn/super-rentals-tutorial/blob/master/src/chapters/12-service-injection.md -->
+
+As promised, we will now work on implementing the share button!
+
+<!-- TODO: add screen shot of the end state -->
+
+While adding the share button, you will learn about:
+
+- Splattributes and the `class` attribute
+- The router service
+- Ember services vs. global variables
+- Mocking services in tests
+
+## Scoping the Feature
+
+In order to be able to share on Twitter, we'll need to make use of the Twitter [Web Intent API](https://developer.twitter.com/en/docs/twitter-for-websites/tweet-button/guides/web-intent.html).
+
+Conveniently, this API doesn't require us to procure any API keys; all we need to do is link to `https://twitter.com/intent/tweet`. This link will prompt the user to compose a new tweet. The API also supports us pre-populating the tweet with some text, hashtag suggestions, or even a link, all through the use of special query params.
+
+For instance, let's say we would like to suggest a tweet with the following content:
+
+```plain
+Check out Grand Old Mansion on Super Rentals! https://super-rentals.example/rentals/grand-old-mansion
+#vacation #travel #authentic #blessed #superrentals via @emberjs
+```
+
+We could open a new page to the <a href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fsuper-rentals.example%2Frentals%2Fgrand-old-mansion&text=Check+out+Grand+Old+Mansion+on+Super+Rentals%21&hashtags=vacation%2Ctravel%2Cauthentic%2Cblessed%2Csuperrentals&via=emberjs" target="_blank" rel="external,nofollow,noopener,noreferrer">following URL</a>:
```

Comment:
Should this link be in markdown?

Reply:
It cannot be in markdown because of the `rel` attribute is not supported